src/daemon/rpmostreed-sysroot.cxx

```diff
@@ -923,10 +917,69 @@ rpmostreed_sysroot_get_repo (RpmostreedSysroot *self)
   return self->repo;
 }
 
+// Fallback method that only authorizes a caller with uid 0.
+// If polkit can't be activated, it's probably not installed.  systemd upstream
+// today falls back to getting the remote pid, then from there gathering the capabilities
+// from /proc to check for CAP_SYS_ADMIN in most cases.  That's nice but also currently
+// racy, and subject to issues when the pid namespace differs.
+//  We'll just be OK with uid == 0.  SELinux should put a stop to processes
+// that run as uid 0 but shouldn't talk to us from working.
+static gboolean
+authorize_method_for_uid0 (GDBusMethodInvocation *invocation, gboolean *out_authorized,
+                           GError **error)
+{
+  *out_authorized = FALSE;
+
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  g_autoptr (GVariant) res = g_dbus_connection_call_sync (
+      g_dbus_method_invocation_get_connection (invocation), "org.freedesktop.DBus",
+      "/org/freedesktop/DBus", "org.freedesktop.DBus", "GetConnectionUnixUser",
+      g_variant_new ("(s)", sender), (GVariantType *)"(u)", G_DBUS_CALL_FLAGS_NONE, -1, NULL,
+      error);
+  if (!res)
+    return FALSE;
+  guint32 uid = 1;
+  g_variant_get (res, "(u)", &uid);
+  *out_authorized = (uid == 0);
+  return TRUE;
+}
+
+// Check whether a method call is authorized without consulting PolicyKit.
+gboolean
+rpmostreed_sysroot_authorize_direct (RpmostreedSysroot *self, GDBusMethodInvocation *invocation,
+                                     gboolean *out_is_authorized, GError **error)
+{
+  *out_is_authorized = FALSE;
+  if (self->on_session_bus)
+    {
+      /* The daemon is on the session bus, running self tests */
+      *out_is_authorized = TRUE;
+      return TRUE;
+    }
```

Comment:
I'm pretty sure the session bus stuff is non-functional today. We should probably just drop whatever bits remain to make this code even simpler/more foolproof.